core/store/src/trie/trie_storage.rs

```diff
@@ -198,10 +202,24 @@ impl TrieCacheInner {
         }
     }
 
+    /// Number of currently cached entries.
     pub fn len(&self) -> usize {
         self.cache.len()
     }
 
+    /// Account consumed memory for a new entry in the cache.
+    pub(crate) fn add_value_of_size(&mut self, len: usize) {
+        let memory_consumed = len as u64 + Self::PER_ENTRY_OVERHEAD;
```

Comment:
nit: it may make sense to make it a function `entry_size(len)` and to avoid duplication. We can even replace `(TrieCacheInner::PER_ENTRY_OVERHEAD + TrieConfig::max_cached_value_size() as u64)` with it while we support old config.